app/src/cli.c

```diff
@@ -368,6 +374,35 @@ parse_record_format(const char *optarg, enum recorder_format *format) {
     return false;
 }
 
+static bool
+parse_codec_profile(const char *optarg, uint32_t *codec_profile) {
+    long value;
+    // long may be 32 bits (it is the case on mingw), so do not use more than
+    // 31 bits (long is signed)
+    bool ok = parse_integer_arg(optarg, &value, true, 0, 0x7FFFFFFF, "codec-profile");
+    if (!ok) {
+        return false;
+    }
+
+    switch (value) {
+        case 0:         // Automatic
+        case 0x01:      // AVCProfileBaseline
+        case 0x02:      // AVCProfileMain
+        case 0x04:      // AVCProfileExtended
+        case 0x08:      // AVCProfileHigh
+        case 0x10:      // AVCProfileHigh10
+        case 0x20:      // AVCProfileHigh422
+        case 0x40:      // AVCProfileHigh444
+        case 0x10000:   // AVCProfileConstrainedBaseline
+        case 0x80000:   // AVCProfileConstrainedHigh
```

Comment:
If only some specific values are authorized, maybe we should pass the strings to the command line parameter instead of an integer?

```
scrcpy --codec-profile=AVCProfileBaseline
```

(the server would still receive an integer)

What do you think?

Reply:
So you want to make a dictionary to map the strings to the integer values?
I just thought it makes the code messier but i guess its better for the users.
I agree.

Reply:
I don't know which is better.

Maybe passing an integer (or hex value) is better in the end (so we can pass anything, that will also automatically work when later scrcpy support AV1 or other codecs).

But in that case, the value must be forwarded as-is, without checks.

Reply:
(oh but passing another value than the "valid" ones will cause problems because level is selected automatically)

Maybe we could just pass `--codec-profile=value` and `--codec-level=value`?